currencies/rate_converter.go

```diff
@@ -160,14 +165,26 @@ func (rc *RateConverter) LastUpdated() time.Time {
 
 // Rates returns current conversions rates
 func (rc *RateConverter) Rates() Conversions {
-	if rc.constantRates != nil {
-		// Converter is not active, returning the constant rates
-		return rc.constantRates
-	}
-	if rates := rc.rates.Load(); rates != nil {
+
+	rates := rc.rates.Load()
+	lastUpdated := rc.lastUpdated.Load()
+
+	if rates != nil && lastUpdated != nil && !rc.CheckStaleRates() {
```

Comment:
How about checking for stale rates during periodic fetching if we were unable to fetch the rates? On the first thought, the approach in this PR seemed okay too but now I am thinking that we'd call `Rates()` for every auction and will therefore go through this piece of code a lot more times as compared to the other approach which I don't think is needed.

The only drawback I can think of the approach that I am suggesting i.e. checking for stale rates during periodic fetching is that if we have say for example, a `fetchInterval` of 45 mins and `staleThreshold` of 120 mins then we will be using the stale rates for 135 mins instead of 120 mins. But, I am not super concerned about this scenario because these are both configurable and we (or anyone else who hosts prebid server) can keep this in mind when they configure these values. Perhaps we can add a warning of some sort where the `StaleRatesSeconds` config option is defined.

Reply:
Good point on the potential performance impact. Under normal circumstances, we shouldn't need to do the time math.

Reply:
@mansinahar and I talked about this offline. I'm going to look at moving the stale rate detection logic to the periodic task and clear the rates when the rates are stale. This logic would then just check for rate presence and use the constant rates if they are nil.